### PR TITLE
API, Core, Flink, Parquet, Spark: Use enhanced for loop

### PR DESCRIPTION
**Pull request by krvikash**
API, Core, Flink, Parquet, Spark: Use enhanced for loop

### CODE REVIEW

File: core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java [hunk 1]
```diff
@@ -114,9 +113,8 @@ public void interleaveValuesFourColumns8ByteOutput(Blackhole blackhole) {
     int outputSize = 8;
     ByteBuffer outputBuffer = ByteBuffer.allocate(outputSize);
 
-    for (int i = 0; i < fourColumnInput.length; i++) {
-      byte[] interleavedBytes =
-          ZOrderByteUtils.interleaveBits(fourColumnInput[i], outputSize, outputBuffer);
+    for (byte[][] bytes : fourColumnInput) {
```
Comment (hililiwei) [inline core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java:116]: bytes -> columnsBinary

File: core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java [hunk 2]
```diff
@@ -75,9 +75,8 @@ public void interleaveValuesFourColumns(Blackhole blackhole) {
     int outputSize = ZOrderByteUtils.PRIMITIVE_BUFFER_SIZE * 4;
     ByteBuffer outputBuffer = ByteBuffer.allocate(outputSize);
 
-    for (int i = 0; i < fourColumnInput.length; i++) {
-      byte[] interleavedBytes =
-          ZOrderByteUtils.interleaveBits(fourColumnInput[i], outputSize, outputBuffer);
+    for (byte[][] bytes : fourColumnInput) {
```
Comment (hililiwei) [inline core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java:78]: bytes -> columnsBinary

**Review by hililiwei — COMMENTED** [review]
Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

**Review by hililiwei — COMMENTED** [review]
Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

**Comment by krvikash** [comment]
> Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

Thanks, @hililiwei for reviewing. Addressed the comments.

**Review by amogh-jahagirdar — APPROVED** [review]
Agreed with @hililiwei, this improves the readability. Thanks for the contribution!

**Review by szehon-ho — APPROVED** [review]

**Comment by szehon-ho** [comment]
Merged, thanks @krvikash , @hililiwei , @amogh-jahagirdar

**Pull request merged**